src/pykeen/datasets/cli.py

```diff
@@ -223,84 +238,125 @@ def verify(dataset: str):
-def expected_metrics(dataset: str, max_triples: Optional[int], log_level: str):
+@force_option
+@click.option("--output-directory", default=PYKEEN_DATASETS, type=pathlib.Path, show_default=True)
+def expected_metrics(
+    dataset: Optional[str],
+    max_triples: Optional[int],
+    min_triples: Optional[int],
+    log_level: str,
+    samples: int,
+    force: bool,
+    output_directory: pathlib.Path,
+):
     """Compute expected metrics for all datasets (matching the given pattern)."""
     logging.getLogger("pykeen").setLevel(level=log_level)
-    directory = PYKEEN_DATASETS
     df_data: List[Tuple[str, str, str, str, float]] = []
-    for _dataset_name, dataset_cls in _iter_datasets(regex_name_filter=dataset, max_triples=max_triples):
+    for _dataset_name, dataset_cls in _iter_datasets(
+        regex_name_filter=dataset, max_triples=max_triples, min_triples=min_triples
+    ):
         dataset_instance = get_dataset(dataset=dataset_cls)
         dataset_name = dataset_resolver.normalize_inst(dataset_instance)
-        d = directory.joinpath(dataset_name, "analysis")
-        d.mkdir(parents=True, exist_ok=True)
-        expected_metrics = dict()
-        for key, factory in dataset_instance.factory_dict.items():
-            if key == "training":
-                additional_filter_triples = None
-            elif key == "validation":
-                additional_filter_triples = dataset_instance.training.mapped_triples
-            elif key == "testing":
-                additional_filter_triples = [
-                    dataset_instance.training.mapped_triples,
-                ]
-                if dataset_instance.validation is None:
-                    click.echo(f"WARNING: {dataset_name} does not have validation triples!")
+        adjustments_directory = output_directory.joinpath(dataset_name, "adjustments")
+        adjustments_directory.mkdir(parents=True, exist_ok=True)
+        expected_metrics_path = adjustments_directory.joinpath("expected_metrics.json")
+        if expected_metrics_path.is_file() and not force:
+            expected_metrics_dict = json.loads(expected_metrics_path.read_text())
+        else:
+            expected_metrics_dict = dict()
+            for key, factory in dataset_instance.factory_dict.items():
+                if key == "training":
+                    additional_filter_triples = None
+                elif key == "validation":
+                    additional_filter_triples = dataset_instance.training.mapped_triples
+                elif key == "testing":
+                    additional_filter_triples = [
+                        dataset_instance.training.mapped_triples,
+                    ]
+                    if dataset_instance.validation is None:
+                        click.echo(f"WARNING: {dataset_name} does not have validation triples!")
+                    else:
+                        additional_filter_triples.append(dataset_instance.validation.mapped_triples)
```

Comment:
duplicate code with training pipeline?

Reply:
i don't think so

Reply:
it looks a bit like https://github.com/pykeen/pykeen/blob/dca8b83272b984e17025ffa9187791ca2b48ed74/src/pykeen/pipeline/api.py#L1255-L1295 but there is more going on there.